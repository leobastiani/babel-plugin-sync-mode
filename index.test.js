const babel = require("@babel/core");
const plugin = require("./index");
const fs = require("fs");
const path = require("path");

const example1 = fs.readFileSync(
  path.resolve(__dirname, "examples", "example1.js"),
  "utf-8"
);

it("works", () => {
  const { code } = babel.transform(example1, { plugins: [plugin] });
  expect(code).toMatchSnapshot();
});
