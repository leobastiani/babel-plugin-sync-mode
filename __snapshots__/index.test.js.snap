// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"async function test() {
  sync: {
    await console.log(\\"I should be awaited 1\\");
    await console.log\`I should be awaited 2\`; // a await goes here

    await console.log( // a await goes here as well
    await (() => {
      console.log(\\"I should not be awaited 3\\");
    })()); // a await goes here

    await console.log( // a await goes here as well
    await (async () => {
      await console.log(\\"I should not be awaited 4\\");
    })());

    async: {
      console.log(\\"I should not be awaited 5\\");

      sync_: {
        await console.log(async () => {
          await console.log(\\"I should be awaited 6\\");
        }, () => {
          console.log(\\"I should not be awaited 7\\");
        });
      }
    }
  }

  console.log(\\"I should not be awaited 8\\");
}

sync: {
  console.log(\\"I should not be awaited 9\\");

  (() => {
    console.log(\\"I should not be awaited 10\\");
  })();

  (async () => {
    await console.log(\\"I should be awaited 111\\");
  })();
}

(async () => {
  console.log(\\"I should not be awaited 112\\");
})();"
`;
